website/docs/guides/migration-guides/migrating-from-v2-to-v3.md

```diff
@@ -0,0 +1,193 @@
+
+- MessageProduceStarted
+- MessageProduceCompleted
+- MessageProduceError
+- MessageConsumeStarted
+- MessageConsumeCompleted
+- MessageConsumeError
+- WorkerStopping
+- WorkerStopped
+- WorkerProcessingEnded
+
+While worker-related events are available through the `IWorker` reference, the message-related events can be subscribed using the `.SubscribeGlobalEvents()` configuration method.
+
+```csharp
+services.AddKafka(
+    kafka => kafka
+        .UseConsoleLog()
+        .SubscribeGlobalEvents(hub =>
+        {
+            hub.MessageConsumeStarted.Subscribe(eventContext => Console.WriteLine("Message Consume Started"));
+
+            hub.MessageConsumeError.Subscribe(eventContext => Console.WriteLine("Message Consume Error"));
+
+            hub.MessageConsumeCompleted.Subscribe(eventContext => Console.WriteLine("Message Consume Completed"));
+
+            hub.MessageProduceStarted.Subscribe(eventContext => Console.WriteLine("Message Produce Started"));
+
+            hub.MessageProduceError.Subscribe(eventContext => Console.WriteLine("Message Produce Error"));
+
+            hub.MessageProduceCompleted.Subscribe(eventContext => Console.WriteLine("Message Produce Completed"));
+        })
+);
+```
+
+### 4.4 Improve Dependency Injection Scope Management ([#428](https://github.com/Farfetch/kafkaflow/issues/428))
+
+Some improvements were made to the way the dependency scopes are managed to provide more fine-grained control over dependency lifecycles. Here are some changes that were made in this context:
+
+- `IMessageContext` now exposes a property `DependencyResolver` which is intrinsically tied to the scope of a single processed message.
+- `IConsumerContext` introduces two properties: `ConsumerDependencyResolver` and `WorkerDependencyResolver`. The former will resolve dependencies tied to the lifecycle of a single consumer, whereas the latter caters to a single worker's lifecycle.
+
+## 5. Improved Worker Distribution Strategy ([#440](https://github.com/Farfetch/kafkaflow/issues/440))
+
+KafkaFlow v2 already provided a method to define the worker to handle a message when consuming using the `IDistributionStrategy` interface. However, the only context provided by this method was the partition key of the message, which sometimes can be insufficient to determine how to distribute messages across workers in more complex scenarios.
+
+With KafkaFlow v3 this interface got renamed to `IWorkerDistributionStrategy` and now the method `.GetWorkerAsync()` receives a `WorkerDistributionContext` structure. Properties like message topic and partition are now part of the context which provides more flexibility when choosing the worker to handle a message.
+
+## Conclusion
+
+Please ensure you review and adapt your codebase according to these changes. If you encounter any issues or need assistance, feel free to reach out to the KafkaFlow community. Thank you for using KafkaFlow!
```

Comment:
Maybe we could add a reference for the [Get In Touch](https://github.com/Farfetch/kafkaflow#get-in-touch) section in the readme.

Reply:
Added the link, please check